Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
```

Comment:
> Or maybe you mean a generic `needs_req_or_dep_with_name?(fooreqdepname)`?

That's what it basically boils down and the method name is certainly very descriptive, but I'm not insisting on generalizing this (unless existing code paths in Homebrew would immediately benefit from such a generalization).